apps/src/dance/ai/DanceAiClient.ts

```diff
@@ -40,66 +42,62 @@ const CachedPalettes: CachedWeightsMapping = UntypedCachedPalettes;
  * }
  */
 export function chooseEffects(
-  emojis: string[],
+  selectedEmojis: string[],
   quality: ChooseEffectsQuality
 ): GeneratedEffect {
-  // Obtain final summed output weight based off input received
-  const outputTypes: CachedWeightsMapping[] = [
-    CachedBackgrounds,
-    CachedForegrounds,
-    CachedPalettes,
-  ];
-  const outputWeights: number[][] = outputTypes.map(function (map) {
-    return calculateOutputWeightsVector(emojis, map);
-  });
-
-  // Sort and slice top scoring options, mapped to their output identifiers (e.g. [[0.25, 'squiggles'], ...])
+  // Obtain final output summed weights based on set of three selected emoji inputs
+  const cachedWeightsMappings: {[key: string]: CachedWeightsMapping} = {
```

Comment:
Also minor - could this be declared outside of this function (alongside `CachedPalettes`, etc) since it's always the same for every function invocation?

Reply:
Nice! And removed the individual mappings variables since no longer needed!